### PR TITLE
fix: empty dora data when run the tasks

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
fix: empty dora data when run the tasks

### Does this close any open issues?
Closes na

### Screenshots
image

image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/dora/tasks/change_lead_time_calculator.go [hunk 1]
```diff
@@ -47,6 +47,12 @@ func CalculateChangeLeadTime(taskCtx plugin.SubTaskContext) errors.Error {
 	db := taskCtx.GetDal()
 	logger := taskCtx.GetLogger()
 	data := taskCtx.GetData().(*DoraTaskData)
+	// Clear previous results from the project
+	deleteSql := fmt.Sprintf("DELETE FROM project_pr_metrics WHERE project_name = '%s'", data.Options.ProjectName)
```
Comment (klesh) [inline backend/plugins/dora/tasks/change_lead_time_calculator.go:51]: SQL injection

File: backend/plugins/dora/tasks/deployment_commits_generator.go [hunk 2]
```diff
@@ -101,30 +101,46 @@ func GenerateDeploymentCommits(taskCtx plugin.SubTaskContext) errors.Error {
...
 		clauses = append(clauses, dal.Where(`p.cicd_scope_id = ?`, data.Options.ScopeId))
+		// Clear previous results from the project
+		deleteSql := `DELETE FROM cicd_deployment_commits
+				WHERE cicd_scope_id IN (
+				SELECT cicd_scope_id
+				FROM (
+					SELECT cdc.cicd_scope_id
+					FROM cicd_deployment_commits cdc
+					LEFT JOIN project_mapping pm ON (pm.table = 'cicd_scopes' AND pm.row_id = cdc.cicd_scope_id)
+				) AS subquery
```
Comment (klesh) [inline backend/plugins/dora/tasks/deployment_commits_generator.go:112]: `project_mapping` should not be needed when using `scopeId`

File: backend/plugins/dora/tasks/deployment_commits_generator.go [hunk 3]
```diff
@@ -101,30 +101,46 @@ func GenerateDeploymentCommits(taskCtx plugin.SubTaskContext) errors.Error {
...
 		// Clear previous results from the project
-		deleteSql := fmt.Sprintf("DELETE cdc FROM cicd_deployment_commits cdc LEFT JOIN project_mapping pm ON (pm.table = 'cicd_scopes' AND pm.row_id = cdc.cicd_scope_id) WHERE pm.project_name = '%s'", data.Options.ProjectName)
+		deleteSql := fmt.Sprintf(`DELETE FROM cicd_deployment_commits
+			WHERE cicd_scope_id IN (
+			SELECT cicd_scope_id
+			FROM (
+				SELECT cdc.cicd_scope_id
+				FROM cicd_deployment_commits cdc
+				LEFT JOIN project_mapping pm ON (pm.table = 'cicd_scopes' AND pm.row_id = cdc.cicd_scope_id)
+				WHERE pm.project_name = '%s'
```
Comment (klesh) [inline backend/plugins/dora/tasks/deployment_commits_generator.go:131]: sql injection

File: backend/plugins/dora/tasks/deployment_generator.go [hunk 4]
```diff
@@ -80,31 +80,48 @@ func GenerateDeployment(taskCtx plugin.SubTaskContext) errors.Error {
 		clauses = append(clauses,
 			dal.Where("p.cicd_scope_id = ?", data.Options.ScopeId),
 		)
+		// Clear previous results from the cicd_scope_id
+		deleteSql := `DELETE FROM cicd_deployments
```
Comment (klesh) [inline backend/plugins/dora/tasks/deployment_generator.go:84]: should not need `project_mapping`

File: backend/plugins/dora/tasks/incident_deploy_connector.go [hunk 5]
```diff
@@ -48,6 +48,12 @@ type simpleCicdDeploymentCommit struct {
 func ConnectIncidentToDeployment(taskCtx plugin.SubTaskContext) errors.Error {
 	db := taskCtx.GetDal()
 	data := taskCtx.GetData().(*DoraTaskData)
+	// Clear previous results from the project
+	deleteSql := fmt.Sprintf("DELETE FROM project_issue_metrics WHERE project_name = '%s'", data.Options.ProjectName)
```
Comment (klesh) [inline backend/plugins/dora/tasks/incident_deploy_connector.go:52]: sql injection

Comment (abeizn) [reply backend/plugins/dora/tasks/incident_deploy_connector.go]: done

File: backend/plugins/dora/tasks/deployment_generator.go (hunk 4 above)
Comment (abeizn) [reply backend/plugins/dora/tasks/deployment_generator.go:83]: done

File: backend/plugins/dora/tasks/deployment_commits_generator.go (hunk 2 above)
Comment (abeizn) [reply backend/plugins/dora/tasks/deployment_commits_generator.go:111]: done

**Review by klesh — APPROVED** [review]

**Pull request merged**